enclave_manager/avalon_enclave_manager/kme/kme_enclave_manager.py

```diff
@@ -192,7 +193,12 @@ def PreProcessWorkOrder(self, **params):
             return wo_response_json["result"]
         else:
             logger.error("Could not preprocess work order at KME")
-            return wo_response_json
+            # For all negative cases, response should have an error field.
+            # Hence constructing JSONRPC error response with
+            # error code and message mapped to KME enclave response
+            err_code = wo_response_json["error"]["code"]
```

Comment:
What about returning wo_response_json itself? Whoever constructing validate error json then return wo_response_json. If it is not valid error jprc response then it needs to fixed at the place where it is consructed.

Reply:
If we return wo_response_json as it is, the JSONRPCDispatcher will internally append "result", "id" and "json-rpc" (duplicate) parameters. This would result in increased complexity at WPE side to check whether the preprocess returned response with result (success case) or error (negative case).